research-software-directory/RSD-as-a-service
# Upgrade front-end dependencies

# Upgrade front-end dependencies

Changes proposed in this pull request:
* Upgrade front-end dependencies: next v13.2.3, d3, jsonwebtoken, dnd-kit, react-testing-library, react-hook-form, react-markdown, tailwind-typography
* Fixed tests required changes after upgrade
* Performed load-tests locally to confirm no (significant) performance loss (see image)

**Important note**
Upgrading material-ui and emotion to the latest version causes fatal app errors I was not able to solve within reasonable time frame. I decided to reverse back to last proper working version of material-ui and emotion. This should be point of attention for the next dependency upgrade.

How to test:
* `make start` to rebuild app
* Examine pages, create/edit software or projects
* `make e2e-tests` to run end to end tests. All tests should pass
* Run frontend unit tests locally. Alternatively, this PR will also run e2e and unit tests

## Load tests (localhost) - no significant performance change

![image](https://user-images.githubusercontent.com/9204081/224071631-4dedd66e-23b9-4dd9-b5e5-4e09984c57b5.png)

PR Checklist:
